tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Actions/Test_ResetSubdomainSolver.cpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "Informer/Tags.hpp"
+#include "Informer/Verbosity.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Actions/ResetSubdomainSolver.hpp"
+#include "ParallelAlgorithms/LinearSolver/Schwarz/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+
+struct DummyOptionsGroup {};
+
+struct SubdomainSolver {
+  void reset() noexcept { is_reset = true; }
+  bool is_reset = false;
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) { p | is_reset; }
+};
+
+template <typename Metavariables>
+struct ElementArray {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = ElementId<1>;
+  using phase_dependent_action_list = tmpl::list<
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Initialization,
+          tmpl::list<ActionTesting::InitializeDataBox<
+              tmpl::list<LinearSolver::Schwarz::Tags::SubdomainSolver<
+                  std::unique_ptr<SubdomainSolver>, DummyOptionsGroup>>>>>,
+      Parallel::PhaseActions<
+          typename Metavariables::Phase, Metavariables::Phase::Testing,
+          tmpl::list<LinearSolver::Schwarz::Actions::ResetSubdomainSolver<
+              DummyOptionsGroup>>>>;
+};
+
+struct Metavariables {
+  using element_array = ElementArray<Metavariables>;
+  using component_list = tmpl::list<element_array>;
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+void test_reset_subdomain_solver(const bool skip_resets) {
```

Comment:
`noexcept`

Reply:
I don't think we have been strict about this in tests at all 😄 Added it though.